core/src/main/java/io/questdb/cutlass/http/HttpConnectionContext.java

```diff
@@ -1034,4 +1005,26 @@ private void shiftReceiveBufferUnprocessedBytes(long start, int receivedBytes) {
         Vect.memmove(recvBuffer, start, receivedBytes);
         LOG.debug().$("peer is slow, waiting for bigger part to parse [multipart]").$();
     }
+
+    @Override
+    protected void doInit() {
+        // the context is obtained from the pool, so we should initialize the memory
+        if (recvBuffer == 0) {
+            // re-read recv buffer size in case the config was reloaded
+            recvBufferSize = configuration.getRecvBufferSize();
+            recvBufferReadSize = Math.min(forceFragmentationReceiveChunkSize, recvBufferSize);
+            recvBuffer = Unsafe.malloc(recvBufferSize, MemoryTag.NATIVE_HTTP_CONN);
+        }
+        // re-read buffer sizes in case the config was reloaded
+        responseSink.of(socket, configuration.getSendBufferSize());
+        headerParser.reopen(configuration.getHttpContextConfiguration().getRequestHeaderBufferSize());
+        multipartContentHeaderParser.reopen(configuration.getHttpContextConfiguration().getMultipartHeaderBufferSize());
+
+        if (socket.supportsTls()) {
```

Comment:
can we move this into `handleClientRecv` please. Check the impl in JavaTls - this is very heavy.

Reply:
i see, init() is no longer called on accept thread...

Reply:
JavaTls is for the http client, and this code has been moved out from connection accept thread already.